datapackage/resource.py

```diff
@@ -247,7 +247,12 @@ def iter(self):
         '''
         result = None
         inline_data = self.metadata.get('data')
-        data_path_or_url = self.metadata.get('path', self.metadata.get('url'))
+        data_path_or_url = self.metadata.get('path')
+        if data_path_or_url is None:
+            data_path_or_url = self.metadata.get('url')
+        else:
+            if self._base_path is not None:
+                data_path_or_url = os.path.join(self._base_path, data_path_or_url)
```

Comment:
This might be easier to write as:

```python
data_path_or_url = self.local_data_path
if data_path_or_url is None:
    data_path_or_url = self.metadata.get('url')
```

Or something along these lines.